### ISSUE TITLE
Pitch move allows to circumvent disable_jump for liquids

### ISSUE THREAD

**Issue by Desour** [minetest/minetest#13598]
##### Minetest version
```
Minetest 5.8.0-dev-c549e84ab
```

##### Summary

If you enable pitch-move, you can swim upwards in nojump liquids.

I'd expect this not to be possible, as `disable_jump` is supposed to be a physical property of the liquid, not a controls altering property for the player.

Note that climbable nodes don't have this issue, as pitch move doesn't happen in climbable nodes.
(Just if you wonder: Making the liquid climbable doesn't have an effect, so you can't use this as a workaround.)

##### Steps to reproduce

* In devetest place `testnodes:liquid_nojump`.
* Jump in.
* Enable pitch move.
* Swim upwards by looking upwards and pressing the forward-move button (default W).

**Comment by srifqi** [comment]
I can confirm in `5.7.0` as well. The pitch move for swimming is first added by PR 7943.

**Comment by Wuzzy2** [comment]
I wanna bet this affects `disable_descend` as well.

**Comment by Desour** [comment]
> I wanna bet this affects `disable_descend` as well.

Yes.

**Comment by JosiahWI** [comment]
You're not jumping, you're just swimming up!

**Comment by Desour** [comment]
> You're not jumping, you're just swimming up!

>I'd expect this not to be possible, as `disable_jump` is supposed to be a physical property of the liquid, not a controls altering property for the player.

**Comment by Wuzzy2** [comment]
To clarify: The purpose of `disable_jump` and `disable_descend` in liquids is to prevent up/down movement by direct player control (i.e. pressing jump or sneak). Therefore, I would argue, pitch move should also not allow a player to go down or up if the respective groups are in place.

Note the "direct player control", this means if the player uses controls to go that direction. These groups do NOT prevent player moving up or down by other means, like falling, `add_player_velocity`, sinking, etc. This is by design; it was always this way for `disable_jump` since it was added for liquids. Yes, `disable_jump` somewhat misnamed here but this is for historic reasons.

In short, the expected behavior is that pitch move doesn't allow you to go down for `disable_descend` nodes and up for `disable_jump` nodes.

I hope this clarifies it.
